sql/2020/06_Third_Parties/distribution_of_3XX_response_body_size.sql

```diff
@@ -0,0 +1,50 @@
+#standardSQL
+# Distribution of response body size by redirected third parties
+# HTTP status codes documentation: https://developer.mozilla.org/en-US/docs/Web/HTTP/Status
+
+WITH requests AS (
+  SELECT
+    _TABLE_SUFFIX AS client,
+    req_host AS host,
+    status,
+    respBodySize AS body_size
+  FROM
+    `httparchive.summary_requests.2020_08_01_*`
+),
+third_party AS (
+  SELECT
+    domain
+  FROM
+    `httparchive.almanac.third_parties`
+  WHERE
+    date = '2020-08-01'
+),
+base AS (
+  SELECT
+    client,
+    domain,
+    IF(status BETWEEN 300 AND 399, 1, 0) AS redirected,
+    body_size
+  FROM
+    requests
+  LEFT JOIN
+    third_party
+  ON
+    NET.HOST(requests.host) = NET.HOST(third_party.domain)
+)
+
+SELECT
+  client,
+  percentile,
+  APPROX_QUANTILES(body_size, 1000)[OFFSET(percentile)] AS approx_redirect_body_size
+FROM
+  base,
+UNNEST(GENERATE_ARRAY(0, 1000, 1)) AS percentile
```

Comment:
@rviscomi The idea was to look into the distribution curve. And I agree it makes sense to simplify it for release.
@simonhearne this will map the main segments body sizes from ~100 - 420 bytes. We could set interesting numbers from the curve to a separate query. What do you plan to include in the chapter?

Reply:
Yeah, most chapters only use those few percentiles because they're easier to visualize and explain to readers. We also have charts in the data viz gallery for distribution values like these. Some chapters also generate histograms but they're a bit trickier to get the bin widths the right size.

Unless @simonhearne has any objections, let's commit the simplified version for each query.

Reply:
@max-ostapenko to keep this moving, I'd suggest committing the suggested changes and we can revisit in a follow-up PR if there are any changes needed.